server/controllers/api-controller.ts

```diff
@@ -1,40 +1,68 @@
 import { APICall } from 'mini-route-loader'
 
 import AppHelper from '../../lib/app-helper'
-import AuthHelper from '../../lib/auth-helper'
-import { EIP712SDK } from '../../lib/EIP712SDK'
-import MongoInterface from '../../lib/mongo-interface'
+import { sendEmail } from '../../lib/mail-sender'
+import MongoInterface, { WebhookReceipt } from '../../lib/mongo-interface'
 
 export default class ApiController {
-  constructor(
-    public mongoInterface: MongoInterface
-  ) {}
+  constructor(public mongoInterface: MongoInterface) {}
 
   ping: APICall = async (req: any, res: any) => {
     return res.status(200).send({ success: true })
   }
 
   /*
-    @notice Generates an attestation using the pre-initialized eip712SDK
-
-    @param recipientAddress The public address of the account to attest to be verified for MarketRegistry
-    @param authToken The authentication token that proves that you are authorized to make this API call. Must exist in the mongo database. 
-
+    @notice Accepts a webhook call from Bloom API 
+ 
   */
-  receiveBloomWebhook: APICall = async (req: any, res: any) => {
+  receiveWebhook: APICall = async (req: any, res: any) => {
+    console.log('received webhook', req)
+
     const inputParams = req.body
 
     const inputs = {
-      recipientAddress: inputParams.recipientAddress,
-      authToken: inputParams.authToken,
+      requestId: inputParams.requestId,
+      user: inputParams.user,
+      template: inputParams.template,
+      profile: inputParams.profile,
+      application: inputParams.application,
+    }
+
+    const receipt: WebhookReceipt = {
+      requestId: inputs.requestId,
+      user: inputs.user,
+      template: inputs.template,
+      profile: inputs.profile,
+      application: inputs.application,
+      createdAt: Date.now(),
     }
 
+    let createdRecord
+    let sentEmail
 
-    console.log('received webhook')
+    try {
+      createdRecord = await this.mongoInterface.WebhookReceiptModel.create(
+        receipt
+      )
+
+      console.log('inserted', createdRecord)
+    } catch (error) {
+      console.error(error)
+    }
+
+    try {
+      sentEmail = await sendEmail(
+        'Bloom API Alert',
+        'A webhook has been received with request_id '.concat(inputs.requestId)
+      )
```

Comment:
Seems like not a lot of data in the email. I assume @udaiparvatha is ok with this?

Reply:
Yes Udai is okay with this, i showed him how to look up the data in mongo for now